third_party/intel/lib/Analysis/DPAS.cpp

```diff
@@ -65,22 +59,11 @@ DPASAnalysis::canUseDPAS(FunctionOpInterface funcOp) const {
     return Result::Maybe;
 
   unsigned threadsPerWarp = cast<IntegerAttr>(threadsPerWarpAttr).getInt();
-  DeviceArch arch = getDeviceArch(mod);
-  if (threadsPerWarp == supportedThreadsPerWarp(arch))
-    return Result::True;
-
-  return Result::False;
-}
-
-unsigned DPASAnalysis::supportedThreadsPerWarp(DeviceArch arch) {
-  switch (arch) {
-  case DeviceArch::PVC:
-    return 16;
-  case DeviceArch::ATS:
-    return 8;
-  default:
-    llvm_unreachable("Unexpected target architecture");
-  }
+  unsigned minSGSize =
+      cast<IntegerAttr>(
+          mod->getAttr(TritonIntelGPUDialect::getMinSGSizeAttrName()))
+          .getInt();
```

Comment:
```suggestion
          mod->getAttr<IntegerAttr>(TritonIntelGPUDialect::getMinSGSizeAttrName())
          .getInt();
```

I think this is possible

Reply:
done